### ISSUE TITLE
Braces behavior using sl_SI (Slovenian) layout on Linux PC

### ISSUE THREAD

**Issue by otalado** [microsoft/vscode#235552]
Type: <b>Bug</b>

Typing the sequence: abc[123] shows as: ab[12]3c in code editor.
Same thing: abc{123} becomes ab{12}3c and abc<123> becomes ab<12>3c, only the round braces are OK. 
For better illustration:
typing [1 works as [1  but only at start of line, 
typing 1] brings ]1,
typing 1[ brings [1. same goes for the {} and <> 

When I try the same thing in a terminal all is OK. Doing the same thing in Windows works OK too.
Braces (except the round ones) on Slovenian keyboards ar two key combinations:
AltGR+ F : [
AltGR+ G: ] 
AltGR+ B: {
AltGR+ N: }
AltGR+ Comma: <
AltGR+ Dot : >
Evidently these are not handled in VSCode the same way on Linux as on Windows.
The same issue you can excpect with Croatian an Serbian(Latin) layouts too.
This I noticed about 6 years ago, but as I do most of my work on Windows I took no action about it. Because this is very annoying bug, I also expected it'll be fixed fast. But ... 

VS Code version: Code 1.95.3 (f1a4fb101478ce6ec82fe9627c43efbf9e98c813, 2024-11-13T14:50:04.152Z)
OS version: Linux arm64 6.6.62+rpt-rpi-2712
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Cortex-A76 (4 x 2800)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off<br>webnn: unavailable_software|
|Load (avg)|0, 1, 1|
|Memory (System)|7.86GB (5.62GB free)|
|Process Argv|--crash-reporter-id ad87f234-1214-4142-a8cb-58d922f732fb|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|xfce|
|XDG_CURRENT_DESKTOP|XFCE|
|XDG_SESSION_DESKTOP|undefined|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
black-formatter|ms-|2024.4.0
debugpy|ms-|2024.12.0
python|ms-|2024.20.0
vscode-pylance|ms-|2024.12.1
reload|nat|0.0.7

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
vscaac:30438847
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupyter:31046869
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc2:31192216
cf971741:31144450
iacca1:31171482
notype1:31157159
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530

```

</details>

**Comment by yanedev** [comment]
Hi, I've been using vsc for 4 years now and I'm programming on Serbian Latin and didn't have any problems. 
AltGR+ F : [
AltGR+ G: ]
AltGR+ B: {
AltGR+ N: }
AltGR+ Comma: <
AltGR+ Dot : >
I'm using this as well and never had a problem. Maybe something else is causing it, not the language...

**Comment by otalado** [comment]
Hi, yanedev,
it is always possible that the problem lays somewhere else. 
But I always have this same problem with VS Code out of the box on every single Linux distribution I tried (and there were quite some) and on x86, x64 , armhf, aarch64  platforms. Also, I tried with Atom editor on x64 Debian 11 a while ago, it did not show any such behavior but it did not favorably compare to VS Code, so I ditched it.
So I have to ask you: 
- did the things worked for you out of the box or did you have to tweak some settings?
- which Linux distribution are you using?
- exactly which keyboard type and keyboard locale are you using?

**Comment by otalado** [comment]
BTW: I totally forgot to mention that any other character which require pressing AltGR on Slovenian standard keyboard are not working too: \, |, ~, ˇ, ^, ˚ e.t.c. Mostly backslash and pipe characters are badly missing. All that works on Windows platforms!

**Comment by otalado** [comment]
I was thinking about the most probable differences between my and yanedv's setup. Serbian Latin uses the same character set as the Slovenian language, so he could use Slovenian keyboard too. 
I decided to install fresh MX Linux on my test machine. This time I installed it with the Slovenian as the OS's language. I hate that some folder names are translated into Slovenian (breaking the standard naming). To fix that, I used _dpkg-reconfigure locales_ to switch back to en_US.UTF-8. After the reboot, I checked VS Code's behavior.
**It worked OK**
Now I must figure out if I can achieve the same thing on some Mint, Armbian and Debian machines preferably without reinstalling.
With that, I'll close the issue. Thanks to yandev, his reply pointed me in the right direction.
